x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integrations.tsx

```diff
@@ -8,55 +8,101 @@
 import React from 'react';
 import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiFormRow, EuiSpacer } from '@elastic/eui';
 import { UseArray, useFormData } from '../../../../shared_imports';
+import type { FormHook, ArrayItem } from '../../../../shared_imports';
 import { RelatedIntegrationsHelpInfo } from './related_integrations_help_info';
 import { RelatedIntegrationFieldRow } from './related_integration_field_row';
 import * as i18n from './translations';
 import { OptionalFieldLabel } from '../optional_field_label';
+import { getFlattenedArrayFieldNames } from '../utils';
 
 interface RelatedIntegrationsProps {
   path: string;
   dataTestSubj?: string;
 }
 
-export function RelatedIntegrations({ path, dataTestSubj }: RelatedIntegrationsProps): JSX.Element {
+function RelatedIntegrationsComponent({
+  path,
+  dataTestSubj,
+}: RelatedIntegrationsProps): JSX.Element {
+  return (
+    <UseArray path={path} initialNumberOfItems={0}>
+      {({ items, addItem, removeItem, form }) => (
+        <RelatedIntegrationsList
+          items={items}
+          addItem={addItem}
+          removeItem={removeItem}
+          path={path}
+          form={form}
+          dataTestSubj={dataTestSubj}
+        />
+      )}
+    </UseArray>
+  );
+}
+
+interface RelatedIntegrationsListProps {
+  items: ArrayItem[];
+  addItem: () => void;
+  removeItem: (id: number) => void;
+  path: string;
+  form: FormHook;
+  dataTestSubj?: string;
+}
+
+const RelatedIntegrationsList = ({
+  items,
+  addItem,
+  removeItem,
+  path,
+  form,
+  dataTestSubj,
+}: RelatedIntegrationsListProps) => {
+  const flattenedFieldNames = getFlattenedArrayFieldNames(form, path);
+
+  /*
+    Not using "watch" for the initial render, to let row components render and initialize form fields.
+    Then we can use the "watch" feature to track their changes.
+  */
+  const hasRenderedInitially = flattenedFieldNames.length > 0;
+  const fieldsToWatch = hasRenderedInitially ? ['index', ...flattenedFieldNames] : [];
```

Comment:
For Related Integrations we don't need to re-render when index pattern change, so we can safely omit it.

```suggestion
  const fieldsToWatch = hasRenderedInitially ? flattenedFieldNames : [];
```